### ISSUE TITLE
A Way to connect to Oracle via LDAP

### ISSUE THREAD

**Issue by yannhyu** [oracle/python-cx_Oracle#353]
### For sample connection string for LDAP, or example of cx_Oracle.makedsn(...) with LDAP connection type in mind:

Are able to make normal connection with something like these:
-      dsn = cx_Oracle.makedsn(p_host, p_port, p_service)
-      con = cx_Oracle.connect(p_username, p_password, dsn, encoding="UTF-8") 

Please provide guidance on doing something similar to the above with LDAP.
Thank you for your help.

**Comment by anthony-tuininga** [comment]
You need to add something like the following in your ``sqlnet.ora`` file:

```
NAMES.DIRECTORY_PATH = (LDAP)
```

and you also need to create an ``ldap.ora`` file that contains something like the following:

```
DIRECTORY_SERVERS = (your-server.your-organization:389:636)
DEFAULT_ADMIN_CONTEXT = "ldap-ou-designation"
DIRECTORY_SERVER_TYPE = OID
```

Once this is in place you can access any of the DSNs that are stored in your LDAP server.

The [documentation](https://cx-oracle.readthedocs.io/en/latest/user_guide/connection_handling.html#optional-oracle-configuration-files) will help in knowing where to place these files. It currently doesn't talk about LDAP, though. You'll have to look at the [general Oracle documentation](https://www.oracle.com/pls/topic/lookup?ctx=dblatest&id=GUID-04F09076-1D04-46DC-BA8C-1B67C1A464FD) for that.

**Comment by cjbj** [comment]
I was given steps once; see https://stackoverflow.com/a/32151099/4799035

**Comment by anthony-tuininga** [comment]
Closing - no activity.

**Comment by exislow** [comment]
I have the same problem on a Mac. Seriously, why it has to be so complicated? Does anybody have more precise directions?

**Comment by cjbj** [comment]
@exislow the topic is technically out of scope for cx_Oracle (which just passes connection strings etc down to Oracle code layers that understand the configuration). I've pinged someone else to take a look.  What _exactly_ are you trying to do?

**Comment by exislow** [comment]
@cjbj thank you for coming back to this issue. Our oracle DB needs to be reached using LDAP. I would like to integrate the Oracle access into my Python script. But this is rather confusing, since this python library is great but do not support LDAP auth connections.

**Comment by cjbj** [comment]
@exislow it does support it, but the layer of code that handles it is in the network authentication layer underneath cx_Oracle.  It is generic to C based Oracle tools and drivers that connect to the DB.

**Comment by exislow** [comment]
Is there a chance to expose this layer, so working with LDAP access is getting more comfortable.
